src/cdk/a11y/interactivity-checker/interactivity-checker.ts

```diff
@@ -132,12 +132,15 @@ export class InteractivityChecker {
    * Gets whether an element can be focused by the user.
    *
    * @param element Element to be checked.
+   * @param includeDisabled When true, considers element to be focusable even if it is disabled.
+   * @param includeInvisible When true, considers element to be focusable even if it is not visible.
    * @returns Whether the element is focusable.
    */
-  isFocusable(element: HTMLElement): boolean {
+  isFocusable(element: HTMLElement, includeDisabled = false, includeInvisible = false): boolean {
```

Comment:
I asked my team for name suggestions for isPotentiallyFocusable. Someone suggested "isFocusableElementType", which seemed nice and descriptive, but overlooks the fact that certain attributes (contenteditable, tabindex) can also make a non-focusable type potentially focusable. Maybe "hasFocusableTypeOrAttributes"?

The option parameters sound like a good backup!

Reply:
Do we actually care about all of the combinations here? (`(true,true)`, `(true,false)`, `(false,true)`, `(false,false)`) or just the `(true,true)` and `(false,false)`?

I actually prefer the flagbag if we care about all of the combinations, otherwise maybe:
- `isCurrentlyFocusable`
- `isFocusableWhenNotDisabled`
- `isFocusableWhenVisible`
- `isFoucsableWhenVisibleAndNotDisabled`
   ^-- bleh, so wordy

Reply:
We care about the default (false, false), and for my use case I care about (false, true) because I want to include invisible elements but not disabled ones. I just added the includeDisabled flag to be consistent, but I don't actually need that flag, nor can I think of any cases off the top of my head where it would be needed.

Reply:
Slept on it, some fresh ideas:
* `hasFocusableIdentity` 
* `hasFocusableQualities`
* `supportsFocus`
* `supportsInteraction`
* `isInteractive`
* `readyToMakeAStageDebut`

Reply:
I find those kind of vague compared to the ones I suggested, I wouldn't know what they do just by looking at them. And I also still like the flagbag, Martin Fowler aside

Reply:
I agree that adding "potentially" is too vague. I think using an optional object parameter containing booleans would be my preference, since it makes it more explicit from the caller code as well as making it easier to modify later

Reply:
Ack- I think the config object with the options is the consensus (though I did prefer `supportsFocus` myself).

So it has been decided!

Reply:
Updated with the crowd favorite config object!